### DISCUSSION TITLE
upgrade selection

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#91] [Q&A: Analysis]
Hi ReEDS team, thank you for your attention!

I have some questions in my head and don't know where to start checking.

(1) Would the retirement schedule be considered a factor in the selection of an upgrade? For example, if a plant retiring in 20 years has a larger chance of being selected for an upgrade than a plant retiring in 2 years?

(2) If an existing plant is selected for the upgrade, does the model extend the retirement year of that plant?

(3) Are there any upgrade settings that can be turned on, so that the simulation upgrades existing capacity (namely, initv(v)) first instead of new capacity (newv(v))?

(4) Is it normal that when Sw_upgrade is turned on, the simulation will try to create a new plant and then upgrade it instead of upgrading the existing capacity directly? For example, I get the result that the simulation tries to create gas-cc plants in 2022 and then upgrade them to gas-cc_gas-cc-ccs around 2026. Meanwhile, from 2022 to 2026, the simulation is still building new gas-cc-ccs plants. Which settings should I check? 

Thank you very much!

**Comment by vcelloho** [comment]
1) Retirement schedule isn't currently a criteria in the selection of a plant, it may appear so because of how existing plant vs new plant heat rates interact with the model
2) Plant upgrades don't extend the life time of the plant, while this is a naive assumption, it's tricky to establish how much longer a plants lifetime would be extended by a generic upgrade.
3) I'm not aware of a switch that would do this, the plant upgrades. The choice to upgrade will be influenced by the heat rate achieved after the upgrade. The upgraded plants heat rate is dependent on the original plants, changing this formula would make existing plants appear more favorable.
4) This doesn't sound abnormal a few things to consider is where these build decisions are occurring, are they in the same location. This may be an issue with model foresight in a sequential solve mode. The cost multipliers intended to give foresight to Gas-CCs on an upcoming CO2 cap are based on a 0 CO2 date. The current investment logic doesn't consider the reduced operations as part of this. Improvements to the cost multipliers could try to capture this. Alternatively running with simultaneous solve will allow you to compare a perfect foresight solution to a sequential solve, if you're encountering computation challenges reducing the number of solve years could help making the model tractable.

**Reply by wanyingw193** [reply]
Thank you! I've got a better idea of where to start.
